workers/task-driver/src/tasks/update_wallet.rs

```diff
@@ -184,12 +183,15 @@ impl Task for UpdateWalletTask {
     type Descriptor = UpdateWalletTaskDescriptor;
 
     async fn new(descriptor: Self::Descriptor, ctx: TaskContext) -> Result<Self, Self::Error> {
-        // Safety check, the new wallet's secret shares must recover the new wallet
-        Self::check_wallet_shares(&descriptor.new_wallet)?;
+        // Pull in the most recent version of the `old_wallet`
+        let old_wallet = ctx
+            .state
+            .get_wallet(&descriptor.old_wallet.wallet_id)?
```

Comment:
Should the descriptor just contain the `old_wallet_id` then instead of the whole `old_wallet`? Looks like that's the only field on the `old_wallet` that we use anywhere

Reply:
good suggestion, done